service/dap/server.go

```diff
@@ -677,6 +710,8 @@ func (s *Server) stopDebugSession(killProcess bool) {
 	if s.debugger == nil {
 		return
 	}
+	// Halting will stop any debugger command that's pending on another
+	// per-request goroutine, hence unblocking that goroutine to wrap-up and exit.
```

Comment:
Technically these additional goroutines could still be not done when this one is. Should we worry about goroutine leaks? Not worth adding a wait group?

Reply:
Looks like I missed this question earlier inside of the resolved conversation.

The goroutine does have a clear path to exit. But if the server is shut down at the same time, the main goroutine could be done before the goroutine checks state, sends stopped event, etc after the running command returns. This is in theory. I haven't actually observed this. If we run this in a test, we could end up with the goroutine still doing things when the test is done. A while back I was advised against this by @eliben. We only really have one such goroutine here with the current design though.